File: Программирование/Лабораторные работы/Lab_2/Код/lab2_2.cs
using System;

namespace lab2_2
{
    class Program
    {
        static void Main(string[] args)
        {
            int ss;
            Console.WriteLine("Введите строку");
            string s = Console.ReadLine();
            ss = s.Length;
            bool flag = true;
            for (int i = 0; i < ss / 2; i++)
            {
                if (Char.ToLower(s[i])!= Char.ToLower(s[ss - i - 1]))
                {
                    Console.WriteLine("Строка не является палиндромом");
                    flag = false;
                    break;
                }
                if (flag)
                {
                    Console.WriteLine("Строка является палиндромом");
                    break;
                }    
            }
        }
    }
}

File: Программирование/Лабораторные работы/Lab_2/Код/lab2_2.pas
program lab2_2;
var
  s,ss: string;
  i: byte; flag: boolean;
begin
   write('Введите строку: ');
   readln(s);
   flag := true;
   for i := 1 to length(s) div 2 do
     if lowercase(s[i]) <> lowercase(s[length(s) - i + 1]) then begin
       writeln('Строка не является палиндромом');
       flag := false;
     break;
     end;
     if flag then
       write('Строка является палиндромом');
     readln;
end.
